lib/ui/painting.dart

```diff
@@ -4302,6 +4302,185 @@ class Canvas extends NativeFieldWrapperClass2 {
 
   /// Draws the given [Image] into the canvas with its top-left corner at the
   /// given [Offset]. The image is composited into the canvas using the given [Paint].
+  ///
+  /// {@tool snippet}
+  ///
+  /// An example usage to draw an asset:
+  ///
+  /// ```dart
+  /// class CustomAssetImage extends StatefulWidget {
+  ///   final String asset;
+  ///
+  ///   const CustomAssetImage({required this.asset});
+  ///
+  ///   @override
+  ///   _CustomAssetImageState createState() => _CustomAssetImageState();
+  /// }
+  ///
+  /// class _CustomAssetImageState extends State<CustomAssetImage> {
+  ///   // ImageInfo provides informations to be used on the image rendering
+  ///   ImageInfo? imageInfo;
+  ///   late final VoidCallback _removeListener;
+  ///
+  ///   @override
+  ///   void initState() {
+  ///     super.initState();
+  ///
+  ///     // ExactAssetImage don't care about the widget context
+  ///     final imageProvider = ExactAssetImage(widget.asset);
+  ///     final stream = imageProvider.resolve(ImageConfiguration.empty);
+  ///
+  ///     // A image stream is used because the image can change over time, like when
+  ///     // there are many frames (e.g. GIFs) or the resource image was mutated
+  ///     final listener = ImageStreamListener(_setImage);
+  ///
+  ///     // Each time a new frame is provided, set [imageInfo] so that the painter
```

Comment:
"Each time a new frame is provided" sounds vaguely like it refers to frames of the application. Maybe "Each time a new image frame is provided (e.g. for an animated GIF)" would be clearer?